src/coreComponents/physicsSolvers/fluidFlow/RegionMultiphaseStatistics.cpp

```diff
@@ -0,0 +1,906 @@
+    regData[Statistics::MAXIMUM_PRESSURE] = MpiWrapper::max( regData[Statistics::MAXIMUM_PRESSURE] );
+    regData[Statistics::MINIMUM_TEMPERATURE] = MpiWrapper::min( regData[Statistics::MINIMUM_TEMPERATURE] );
+    regData[Statistics::MAXIMUM_TEMPERATURE] = MpiWrapper::max( regData[Statistics::MAXIMUM_TEMPERATURE] );
+
+    regData[Statistics::AVERAGE_PRESSURE] = MpiWrapper::sum( regData[Statistics::AVERAGE_PRESSURE] );
+    regData[Statistics::AVERAGE_TEMPERATURE] = MpiWrapper::sum( regData[Statistics::AVERAGE_TEMPERATURE] );
+
+    for( integer propIndex = Statistics::PHASE_PORE_VOLUME; propIndex < Statistics::END; ++propIndex )
+    {
+      regData[propIndex] = MpiWrapper::sum( regData[propIndex] );
+    }
+
+    // Actually calculate averages
+    real64 const invStaticPoreVolume = safeInverse( regData[Statistics::STATIC_PORE_VOLUME] );
+    regData[Statistics::AVERAGE_PRESSURE] *= invStaticPoreVolume;
+    regData[Statistics::AVERAGE_TEMPERATURE] *= invStaticPoreVolume;
+
+    // Calculate average density and viscosity using cell count as weight
+    for( integer phaseIndex = 0; phaseIndex < phaseCount; ++phaseIndex )
+    {
+      real64 const invCellCount = safeInverse( regData[Statistics::CELL_COUNT] );
+      regData[Statistics::PHASE_DENSITY + phaseIndex] *= invCellCount;
+      regData[Statistics::PHASE_VISCOSITY + phaseIndex] *= invCellCount;
+    }
+  }
+
+  if( m_writeCSV > 0 && MpiWrapper::commRank() == 0 )
+  {
+    array1d< real64 > propColumns;
+    populateColumns( time, propColumns, regionStatistics );
+    std::ofstream outputFile( m_outputDir + "/" + viewKeyStruct::fileNameString() + ".csv", std::ios_base::app );
+    writeArray( outputFile, propColumns );
+    outputFile.close();
+  }
+}
+
+void RegionMultiphaseStatistics::initializeFile() const
+{
+  string const fluidModelName = m_solver->referenceFluidModelName();
+
+  DomainPartition const & domain = getGroupByPath< DomainPartition >( "/Problem/domain" );
+  ConstitutiveManager const & constitutiveManager = domain.getConstitutiveManager();
+  MultiFluidBase const & fluidModel = constitutiveManager.getConstitutiveRelation< MultiFluidBase >( fluidModelName );
+  auto fluidComponentNames = fluidModel.componentNames();
+  auto fluidPhaseNames = fluidModel.phaseNames();
+
+  integer const phaseCount = m_solver->numFluidPhases();
+  integer const componentCount = m_solver->numFluidComponents();
+  integer const regionCount = m_regionIdentifiers.size();
+  auto regionNames = m_regionNames.toView();
+
+  string_array propNames;
+  string_array regNames;
+  string_array propUnits;
+  string_array phaseNames;
+  string_array compNames;
+
+  auto addColumn = [&]( string const propName,
+                        string const propUnit,
+                        integer const regionIndex=-1,
+                        integer const phaseIndex=-1,
+                        integer const compIndex=-1 )
+  {
+    propNames.emplace_back( propName );
+    propUnits.emplace_back( propUnit );
```

Comment:
To manage the units, you can use `common/Units.hpp`

```suggestion
  auto addColumn = [&]( string const propName,
                        units::Unit const propUnit,
                        integer const regionIndex=-1,
                        integer const phaseIndex=-1,
                        integer const compIndex=-1 )
  {
    propNames.emplace_back( propName );
    propUnits.emplace_back( GEOS_FMT( "[{}]", getSymbol( propUnit ) ) );
```